app/scripts/lib/nonce-tracker.js

```diff
@@ -25,21 +26,13 @@ class NonceTracker {
     await this._globalMutexFree()
     // await lock free, then take lock
     const releaseLock = await this._takeMutex(address)
-    // calculate next nonce
-    // we need to make sure our base count
-    // and pending count are from the same block
-    const currentBlock = await this._getCurrentBlock()
-    const pendingTransactions = this.getPendingTransactions(address)
-    const pendingCount = pendingTransactions.length
-    assert(Number.isInteger(pendingCount), `nonce-tracker - pendingCount is not an integer - got: (${typeof pendingCount}) "${pendingCount}"`)
-    const baseCountHex = await this._getTxCount(address, currentBlock)
-    const baseCount = parseInt(baseCountHex, 16)
-    assert(Number.isInteger(baseCount), `nonce-tracker - baseCount is not an integer - got: (${typeof baseCount}) "${baseCount}"`)
-    const nextNonce = baseCount + pendingCount
+    const localNextNonce = this._getLocalNextNonce(address)
+    const nonceDetails = await this._getNetworkNonceAndDetails(address)
+    const networkNonce = nonceDetails.networkNonce
+    const nextNonce = Math.max(networkNonce, localNextNonce)
```

Comment:
There's some disparity behind how these two values include the pending transactions into their values, I think it obscures the complexity of the logic happening here.

One way that might be a simpler way to reason about this would be to entirely remove `pending` from both `networkNonce` and `localNextNonce`, and instead do a `Max(nextNetworkNonce, nextLocalConfirmedNonce, nextPendingNonce)`.

This way, we would default to whatever was the next valid nonce for each of these three scenarios, and the logic for calculating each could be isolated.